launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java

```diff
@@ -59,6 +59,18 @@ public void testClusterCmdBuilder() throws Exception {
   }
 
   @Test
+  public void testCliHelpAndNoArg() throws Exception {
+    List<String> sparkSubmitArgs = Arrays.asList(parser.HELP);
+    Map<String, String> env = new HashMap<>();
+    List<String> cmd = buildCommand(sparkSubmitArgs, env);
+    assertTrue("--help should be contained in the final cmd.", cmd.contains(parser.HELP));
+
+    List<String> sparkEmptyArgs = Arrays.asList("");
+    cmd = buildCommand(sparkSubmitArgs, env);
```

Comment:
Oh, and BTW, you should use `Collections.emptyList()`, because you don't really have an empty args list, but an args list with a single argument that happens to be an empty string.